test/opt/opextinst_forward_ref_fixup_pass_test.cpp

```diff
@@ -0,0 +1,365 @@
+                   %4 = OpString "// RUN: %dxc -T cs_6_0 %s -E main -spirv -fspv-target-env=vulkan1.1 -fspv-debug=vulkan-with-source | FileCheck %s
+
+class A {
+  void foo() {
+  }
+};
+
+[numthreads(1, 1, 1)]
+void main() {
+  A a;
+  a.foo();
+}
+"
+                   %5 = OpString "A"
+                   %6 = OpString "A.foo"
+                   %7 = OpString ""
+                   %8 = OpString "this"
+                   %9 = OpString "main"
+                  %10 = OpString "a"
+                  %11 = OpString "d59ae9c2"
+                  %12 = OpString " -E main -T cs_6_0 -spirv -fspv-target-env=vulkan1.1 -fspv-debug=vulkan-with-source -Vd -Qembed_debug"
+                        OpName %main "main"
+                        OpName %A "A"
+                %void = OpTypeVoid
+                %uint = OpTypeInt 32 0
+              %uint_1 = OpConstant %uint 1
+              %uint_4 = OpConstant %uint 4
+              %uint_5 = OpConstant %uint 5
+              %uint_0 = OpConstant %uint 0
+              %uint_3 = OpConstant %uint 3
+              %uint_7 = OpConstant %uint 7
+            %uint_288 = OpConstant %uint 288
+              %uint_9 = OpConstant %uint 9
+             %uint_13 = OpConstant %uint 13
+             %uint_10 = OpConstant %uint 10
+                  %40 = OpTypeFunction %void
+             %uint_12 = OpConstant %uint 12
+                   %A = OpTypeStruct
+         %_ptr_Function_A = OpTypePointer Function %A
+             %uint_11 = OpConstant %uint 11
+                  %15 = OpExtInst %void %1 DebugExpression
+                  %16 = OpExtInst %void %1 DebugSource %3 %4
+                  %17 = OpExtInst %void %1 DebugCompilationUnit %uint_1 %uint_4 %16 %uint_5
+                  %21 = OpExtInst %void %1 DebugTypeComposite %5 %uint_0 %16 %uint_3 %uint_7 %17 %5 %uint_0 %uint_3 %25
+                  %26 = OpExtInst %void %1 DebugTypeFunction %uint_3 %void %21
+                  %25 = OpExtInst %void %1 DebugFunction %6 %26 %16 %uint_4 %uint_3 %21 %7 %uint_3 %uint_4
+                  %27 = OpExtInst %void %1 DebugLocalVariable %8 %21 %16 %uint_4 %uint_3 %25 %uint_288 %uint_1
+                  %29 = OpExtInst %void %1 DebugTypeFunction %uint_3 %void
+                  %30 = OpExtInst %void %1 DebugFunction %9 %29 %16 %uint_9 %uint_1 %17 %7 %uint_3 %uint_9
+                  %32 = OpExtInst %void %1 DebugLexicalBlock %16 %uint_9 %uint_13 %30
+                  %34 = OpExtInst %void %1 DebugLocalVariable %10 %21 %16 %uint_10 %uint_5 %32 %uint_4
+                  %36 = OpExtInst %void %1 DebugEntryPoint %30 %17 %11 %12
+                  %37 = OpExtInst %void %1 DebugInlinedAt %uint_11 %32
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugExpression
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugSource
+; CHECK:       {{.*}} = OpExtInst %void %1 DebugCompilationUnit
+; CHECK:       {{.*}} = OpExtInstWithForwardRefsKHR %void {{.*}} DebugTypeComposite
```

Comment:
This check seems incomplete. You want to also make sure that the old OpExtInst has been removed. If that is covered by the validation, then add a comment saying do.

Reply:
Done, also modified the opposite side to make sure the old OpExtInstWithForwardRefs wasn't there.